packages/opentelemetry-exporter-stackdriver-trace/README.md

```diff
@@ -0,0 +1,98 @@
+# OpenTelemetry Stackdriver Trace Exporter
+[![Gitter chat][gitter-image]][gitter-url]
+[![NPM Published Version][npm-img]][npm-url]
+[![dependencies][dependencies-image]][dependencies-url]
+[![devDependencies][devDependencies-image]][devDependencies-url]
+[![Apache License][license-image]][license-image]
+
+OpenTelemetry Stackdriver Trace Exporter allows the user to send collected traces to Stackdriver.
+
+[Stackdriver Trace](https://cloud.google.com/trace) is a distributed tracing system. It helps gather timing data needed to troubleshoot latency problems in microservice architectures. It manages both the collection and lookup of this data.
+
+## Setup
+
+Stackdriver Trace is a managed service provided by Google Cloud Platform.
+
+### Installation
+
+Install the npm package `@opentelemetry/exporter-stackdriver-trace`
+
+```shell
+$ npm install @opentelemetry/exporter-stackdriver-trace
+```
+
+### Authentication
+
+The Stackdriver Trace exporter supports authentication using service accounts. These can either be defined in a keyfile (usually called `service_account_key.json` or similar), or by the environment. If your application runs in a GCP environment, such as Compute Engine, you don't need to provide any application credentials. The client library will find the credentials by itself. For more information, go to <https://cloud.google.com/docs/authentication/>.
+
+#### Service account key
+
+If you are not running in a GCP environment, you will need to give the service account credentials to the exporter.
+
+```js
+const { StackdriverTraceExporter } = require('@opentelemetry/exporter-stackdriver-trace');
+
+const exporter = new StackdriverTraceExporter({
+  serviceName: 'your-application-name',
+
+  /** option 1. provide a service account key json */
+  keyFile: './service_account_key.json',
+  keyFileName: './service_account_key.json',
+
+  /** option 2. provide credentials directly */
+  credentials: {
+    client_email: string,
+    private_key: string,
+  },
+});
+```
+
+## Usage
+
+Install the exporter on your application and pass the service name.
+
+```js
+const { StackdriverTraceExporter } = require('@opentelemetry/exporter-stackdriver-trace');
+
+const exporter = new StackdriverTraceExporter({
+  serviceName: 'your-application-name',
```

Comment:
`projectId` is mandatory param, we should put it here in the example.

Reply:
projectId is not mandatory. The client may get it from the environment in GCP

Reply:
There are a few places auth can come from. env vars, the service account key json, or in worst case the service account key can be used to make an api call to gcp to get the current project id.

Reply:
Can you please mark this as resolved if you are happy with the answer?

Reply:
+1 to "projectId is not mandatory" it should be inferred from the credentials most of the time and only rarely need to be specified explicitly. It actually makes things much easier to not have to specify it because then you can deploy the same source code to different GCP projects (e.g. test and production) and they will both write traces to the right places without you needing to manage project ID config!